purchase_order_analytic_search/models/purchase_order.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2015 ForgeFlow S.L.
+# Copyright 2018 Camptocamp
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = "purchase.order"
+
+    @api.depends("order_line.account_analytic_id")
+    def _compute_analytic_accounts(self):
+        for purchase in self:
+            purchase.account_analytic_ids = purchase.mapped(
+                "order_line.account_analytic_id"
+            )
+
+    @api.model
+    def _search_analytic_accounts(self, operator, value):
+        po_line_obj = self.env["purchase.order.line"].filtered(
```

Comment:
Maybe it is best to define a base_domain and then use the expression.add to extend the domain?